File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    col = board[0]
    row = board[1]
    result = [0, 0]
    for key in keyinput:
        if key == "left" and result[0] - 1 >= -(col // 2):
            result[0] -= 1
        elif key == "right" and result[0] + 1 <= (col // 2):
            result[0] += 1
        elif key == "up" and result[1] + 1 <= (row // 2):
            result[1] += 1
        elif key == "down" and result[1] - 1 >= -(row // 2):
            result[1] -= 1
    return result

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<string> keyinput, vector<int> board) {
    
    int col = board[0] / 2;
    int row = board[1] / 2;
    vector<int> result = {0, 0};
    
    for (int i = 0; i < keyinput.size(); i++) {
        if (keyinput[i] == "left" && (result[0] - 1) >= -col) result[0] -= 1;
        else if (keyinput[i] == "right" && (result[0] + 1) <= col) result[0] += 1;
        else if (keyinput[i] == "up" && (result[1] + 1) <= row) result[1] += 1;
        else if (keyinput[i] ==  "down" && (result[1] - 1) >= -row) result[1] -= 1;
    }
    return result;
}